jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/shortestpath/DijkstraDistance.java

```diff
@@ -66,10 +66,11 @@
 {
     protected Hypergraph<V,E> g;
     protected Function<? super E,? extends Number> nev;
-    protected Map<V,SourceData> sourceMap;   // a map of source vertices to an instance of SourceData
+    protected LinkedHashMap<V,SourceData> sourceMap;   // a map of source vertices to an instance of SourceData
```

Comment:
Sorry if I missed a discussion of this, but is the cost (of a possible virtual lookup when declaring as the most general type) really a factor with modern JVMs?

Reply:
If it _is_ a factor on modern JVMs, I've personally not heard of it. :/

IIRC, it's more idiomatic to keep the base type here as `Map`, i.e. whatever interface is most generic, or as high up the type hierarchy as possible, that has all the methods one needs.

Just my 2 cents.